src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java

```diff
@@ -0,0 +1,54 @@
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadoctagcontinuationindentation;
+
+import java.io.Serializable;
+
+/**
+ * Config:
+ * offset = 4
+ */
+public class InputJavadocTagContinuationIndentationBlockTag {
+
+    /**
+     * Example from issue 5711.
+     * Returns the value represented by the specified string of the specified
+     * type. Returns 0 for types other than float, double, int, and long.
+     * @param text the string to be parsed.
+     * @param type the token type of the text. Should be a constant of
+     * {@link TokenTypes}. // violation
+     * @return the double value represented by the string argument.
+     */
+    public static double parseDouble(String text, int type) {
+        return 0;
+    }
+
+    /**
+     * Javadoc.
+     *
+     * @param x this line is normal
+     * {@code this} line is wrongly indented // violation
+     */
```

Comment:
> Please add some more complicated test cases, such as:
> 
> ```java
>     /**
>      * Javadoc
>      * @param args  // ok
>      * {@code this} line is not correctly indented // violation
>      *    {@code this}  // ok
>      * <pre>this line is not correctly indented</pre> // violation
>      */
> ```
> 
> Also, it would be good to show violations for html tags and other problem areas for sequential lines, such as:
> 
> ```java
>     /**
>      * Javadoc
>      * @param args  // ok
>      * {@code this} line is not correctly indented // violation
>      * {@code this} line is not correctly indented // violation
>      * <pre>this line is not correctly indented</pre> // violation
>      */
> ```

Ok, added these two cases along with the regression case. Do I need to come up with more?

Reply:
> There is a good test case with html list in diff report, too: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/javadoctagcont-blocktag_2021170726/reports/diff/apache-ant/xref/home/runner/work/checkstyle/checkstyle/contribution/checkstyle-tester/repositories/apache-ant/src/main/org/apache/tools/ant/taskdefs/AbstractCvsTask.java.html#L749

One thing I found that could be slightly unintuitive is that although a violation is printed for line 749, no violation is generated for lines 750-761. This is because our `getAllNewlineNodes` only searches the 'top-level' `DetailNode` objects, and does not look inside tokens such as `HTML_ELEMENT` which could themselves contain newlines.

Below is the Javadoc tree structure for the `InputJavadocTagContinuationIndentationBlockTag` class. A `HTML_ELEMENT` begins at [84:7] and contains all the subsequent newline tokens such as [84:11], [85:11], [86:18].
```
    |--METHOD_DEF -> METHOD_DEF [98:4]
    |   |--MODIFIERS -> MODIFIERS [98:4]
    |   |   |--BLOCK_COMMENT_BEGIN -> /* [78:4]
    |   |   |   |--COMMENT_CONTENT -> *\r\n     * Long javadoc from regression test case (apache-ant).\r\n     *\r\n     * @param c the command line which will be configured\r\n     * if the commandline is initially null, the function is a noop\r\n     * otherwise the function append to the commandline arguments concerning\r\n     * <ul>\r\n     * <li>\r\n     * cvs package\r\n     * </li>\r\n     * <li>\r\n     * compression\r\n     * </li>\r\n     * <li>\r\n     * quiet or reallyquiet\r\n     * </li>\r\n     * <li>cvsroot</li>\r\n     * <li>noexec</li>\r\n     * </ul>\r\n      [78:6]
    |   |   |   |   `--JAVADOC -> JAVADOC [78:7]
    |   |   |   |       |--NEWLINE -> \r\n [78:7]
    |   |   |   |       |--LEADING_ASTERISK ->      * [79:0]
    |   |   |   |       |--TEXT ->  Long javadoc from regression test case (apache-ant). [79:6]
    |   |   |   |       |--NEWLINE -> \r\n [79:59]
    |   |   |   |       |--LEADING_ASTERISK ->      * [80:0]
    |   |   |   |       |--NEWLINE -> \r\n [80:6]
    |   |   |   |       |--LEADING_ASTERISK ->      * [81:0]
    |   |   |   |       |--WS ->   [81:6]
    |   |   |   |       |--JAVADOC_TAG -> JAVADOC_TAG [81:7]
    |   |   |   |       |   |--PARAM_LITERAL -> @param [81:7]
    |   |   |   |       |   |--WS ->   [81:13]
    |   |   |   |       |   |--PARAMETER_NAME -> c [81:14]
    |   |   |   |       |   |--WS ->   [81:15]
    |   |   |   |       |   `--DESCRIPTION -> DESCRIPTION [81:16]
    |   |   |   |       |       |--TEXT -> the command line which will be configured [81:16]
    |   |   |   |       |       |--NEWLINE -> \r\n [81:57]
    |   |   |   |       |       |--LEADING_ASTERISK ->      * [82:0]
    |   |   |   |       |       |--TEXT ->  if the commandline is initially null, the function is a noop [82:6]
    |   |   |   |       |       |--NEWLINE -> \r\n [82:67]
    |   |   |   |       |       |--LEADING_ASTERISK ->      * [83:0]
    |   |   |   |       |       |--TEXT ->  otherwise the function append to the commandline arguments concerning [83:6]    |   |   |   |       |       |--NEWLINE -> \r\n [83:76]
    |   |   |   |       |       |--LEADING_ASTERISK ->      * [84:0]
    |   |   |   |       |       |--TEXT ->   [84:6]
    |   |   |   |       |       |--HTML_ELEMENT -> HTML_ELEMENT [84:7]
    |   |   |   |       |       |   `--HTML_TAG -> HTML_TAG [84:7]
    |   |   |   |       |       |       |--HTML_ELEMENT_START -> HTML_ELEMENT_START [84:7]
    |   |   |   |       |       |       |   |--START -> < [84:7]
    |   |   |   |       |       |       |   |--HTML_TAG_NAME -> ul [84:8]
    |   |   |   |       |       |       |   `--END -> > [84:10]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [84:11]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [85:0]
    |   |   |   |       |       |       |--TEXT ->   [85:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [85:7]
    |   |   |   |       |       |       |   `--LI -> LI [85:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [85:7]
    |   |   |   |       |       |       |       |   |--START -> < [85:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [85:8]
    |   |   |   |       |       |       |       |   `--END -> > [85:10]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [85:11]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [86:0]
    |   |   |   |       |       |       |       |--TEXT ->  cvs package [86:6]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [86:18]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [87:0]
    |   |   |   |       |       |       |       |--TEXT ->   [87:6]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [87:7]
    |   |   |   |       |       |       |           |--START -> < [87:7]
    |   |   |   |       |       |       |           |--SLASH -> / [87:8]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [87:9]
    |   |   |   |       |       |       |           `--END -> > [87:11]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [87:12]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [88:0]
    |   |   |   |       |       |       |--TEXT ->   [88:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [88:7]
    |   |   |   |       |       |       |   `--LI -> LI [88:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [88:7]
    |   |   |   |       |       |       |       |   |--START -> < [88:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [88:8]
    |   |   |   |       |       |       |       |   `--END -> > [88:10]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [88:11]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [89:0]
    |   |   |   |       |       |       |       |--TEXT ->  compression [89:6]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [89:18]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [90:0]
    |   |   |   |       |       |       |       |--TEXT ->   [90:6]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [90:7]
    |   |   |   |       |       |       |           |--START -> < [90:7]
    |   |   |   |       |       |       |           |--SLASH -> / [90:8]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [90:9]
    |   |   |   |       |       |       |           `--END -> > [90:11]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [90:12]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [91:0]
    |   |   |   |       |       |       |--TEXT ->   [91:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [91:7]
    |   |   |   |       |       |       |   `--LI -> LI [91:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [91:7]
    |   |   |   |       |       |       |       |   |--START -> < [91:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [91:8]
    |   |   |   |       |       |       |       |   `--END -> > [91:10]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [91:11]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [92:0]
    |   |   |   |       |       |       |       |--TEXT ->  quiet or reallyquiet [92:6]
    |   |   |   |       |       |       |       |--NEWLINE -> \r\n [92:27]
    |   |   |   |       |       |       |       |--LEADING_ASTERISK ->      * [93:0]
    |   |   |   |       |       |       |       |--TEXT ->   [93:6]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [93:7]
    |   |   |   |       |       |       |           |--START -> < [93:7]
    |   |   |   |       |       |       |           |--SLASH -> / [93:8]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [93:9]
    |   |   |   |       |       |       |           `--END -> > [93:11]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [93:12]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [94:0]
    |   |   |   |       |       |       |--TEXT ->   [94:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [94:7]
    |   |   |   |       |       |       |   `--LI -> LI [94:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [94:7]
    |   |   |   |       |       |       |       |   |--START -> < [94:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [94:8]
    |   |   |   |       |       |       |       |   `--END -> > [94:10]
    |   |   |   |       |       |       |       |--TEXT -> cvsroot [94:11]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [94:18]
    |   |   |   |       |       |       |           |--START -> < [94:18]
    |   |   |   |       |       |       |           |--SLASH -> / [94:19]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [94:20]
    |   |   |   |       |       |       |           `--END -> > [94:22]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [94:23]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [95:0]
    |   |   |   |       |       |       |--TEXT ->   [95:6]
    |   |   |   |       |       |       |--HTML_ELEMENT -> HTML_ELEMENT [95:7]
    |   |   |   |       |       |       |   `--LI -> LI [95:7]
    |   |   |   |       |       |       |       |--LI_TAG_START -> LI_TAG_START [95:7]
    |   |   |   |       |       |       |       |   |--START -> < [95:7]
    |   |   |   |       |       |       |       |   |--LI_HTML_TAG_NAME -> li [95:8]
    |   |   |   |       |       |       |       |   `--END -> > [95:10]
    |   |   |   |       |       |       |       |--TEXT -> noexec [95:11]
    |   |   |   |       |       |       |       `--LI_TAG_END -> LI_TAG_END [95:17]
    |   |   |   |       |       |       |           |--START -> < [95:17]
    |   |   |   |       |       |       |           |--SLASH -> / [95:18]
    |   |   |   |       |       |       |           |--LI_HTML_TAG_NAME -> li [95:19]
    |   |   |   |       |       |       |           `--END -> > [95:21]
    |   |   |   |       |       |       |--NEWLINE -> \r\n [95:22]
    |   |   |   |       |       |       |--LEADING_ASTERISK ->      * [96:0]
    |   |   |   |       |       |       |--TEXT ->   [96:6]
    |   |   |   |       |       |       `--HTML_ELEMENT_END -> HTML_ELEMENT_END [96:7]
    |   |   |   |       |       |           |--START -> < [96:7]
    |   |   |   |       |       |           |--SLASH -> / [96:8]
    |   |   |   |       |       |           |--HTML_TAG_NAME -> ul [96:9]
    |   |   |   |       |       |           `--END -> > [96:11]
    |   |   |   |       |       |--NEWLINE -> \r\n [96:12]
    |   |   |   |       |       `--TEXT ->       [97:0]
    |   |   |   |       `--EOF -> <EOF> [97:5]
    |   |   |   `--BLOCK_COMMENT_END -> */ [97:5]
```

If this behavior is intended, then it should probably be better documented in the usage examples so that it is not mistaken for a bug. Otherwise, if this behavior is unintended then the fix is to modify `getAllNewlineNodes` to check for 'deep' `NEWLINE` tokens. In either case, I think it can be moved to a new issue so that this PR can focus on the bug it was originally intended to fix.

Reply:
> If this behavior is intended, then it should probably be better documented in the usage examples so that it is not mistaken for a bug. Otherwise, if this behavior is unintended then the fix is to modify getAllNewlineNodes to check for 'deep' NEWLINE tokens. In either case, I think it can be moved to a new issue so that this PR can focus on the bug it was originally intended to fix.

I agree on all counts, but it is good that we discovered this.  @wltan can you please create an issue?

Reply:
Opened #9315